AirCasting/Graph/GraphView.swift

```diff
@@ -36,35 +36,92 @@ struct GraphView<StatsViewModelType>: View where StatsViewModelType: StatisticsC
+                                        graphStatsDataSource?.dateRange = range
+                                        statsContainerViewModel?.adjustForNewData()
+                                      }
+                                StatisticsContainerView(statsContainerViewModel: statsContainerViewModel,
+                                                        threshold: threshold)
+                            }
 
-                }
-                NavigationLink(destination: HeatmapSettingsView(changedThresholdValues: threshold.rawThresholdsBinding)) {
-                    EditButtonView()
-                }
-                .padding(.horizontal)
-                .padding(.top)
-                
-                ThresholdsSliderView(threshold: threshold)
-                    .padding()
-                    // Fixes labels covered by tabbar
-                    .padding(.bottom)
+                        }
+                        HStack() {
+                            startTimeText
+                            Spacer()
+                            endTimeText
+                        }
+                        .foregroundColor(.aircastingGray)
+                        .padding(.horizontal, 5)
+                        NavigationLink(destination: HeatmapSettingsView(changedThresholdValues: threshold.rawThresholdsBinding)) {
+                            EditButtonView()
+                        }
+                        .padding(.horizontal)
+                        .padding(.top)
+                        
+                        ThresholdsSliderView(threshold: threshold)
+                            .padding()
+                            // Fixes labels covered by tabbar
+                            .padding(.bottom)
+                    }
             }
             Spacer()
         }
         .navigationBarTitleDisplayMode(.inline)
     }
+    
+    var startTimeText: some View {
+        startTime()
+    }
+    
+    var endTimeText: some View {
+        endTime()
+    }
+    
+    func procceding(session: SessionEntity) -> Bool {
+        var showGraph = false
+        guard session.allStreams != nil else { return false }
+        (session.allStreams!.forEach({ stream in
+            (stream.allMeasurements != []) ? (showGraph = true) : (showGraph = false)
+        }))
+        if showGraph {
+            return true
+        } else {
+            return false
+        }
+    }
+    
+    func startTime() -> Text {
+            let formatter = DateFormatter()
+            formatter.locale = Locale(identifier: "en_US")
+        if !(session.isMobile && session.isActive) {
+            formatter.timeZone =  TimeZone.init(abbreviation: "UTC")
+        }
+            formatter.dateFormat = "HH:mm"
+            guard let start = session.startTime else { return Text("") }
+     
+            let string = formatter.string(from: start)
+            return Text(string)
+        }
+    
+    func endTime() -> Text {
```

Comment:
in this case repeated parts of `startTime()` should be extracted and reused instead of copy/pasted